CRVResponse/src/CrvPhotonGenerator_module.cc

```diff
@@ -181,7 +265,23 @@ namespace mu2e
           double timeOffset = _timeOffsets.totalTimeOffset(step.simParticle());
           double t1 = step.startTime()+timeOffset;
           double t2 = step.endTime()+timeOffset;
-          if(t1<_startTime) continue;   //Ignore this StepPoint to reduce computation time.
+          if(isnan(t1) || isnan(t2)) continue;  //This situation was observed once. Not sure how it happened.
+
+          //see explanation above
+          //On-spill: No photons in the blind time between before digitizationStart and up to 55ns after 0
+          //(which will moved past the microbunch end after time folding) but allow a little bit more for the crvSteps
+          //-record CrvSteps before digitizationEnd-microBunchPeriod (i.e. 1750ns-1695ns=55ns)
+          //-record CrvSteps after digitizationStart-crvStepMargin (i.e. 400ns-50ns=350ns)
+          //Off-spill: Only photons within eventWindow, but allow a little bit more for the crvSteps
+          //-record CrvSteps after eventwindowStart-crvStepMargin until eventWindowEnd
+          if(spillType==EventWindowMarker::SpillType::onspill)
+          {
+            if(t1>digitizationEnd-_microBunchPeriod && t2<digitizationStart-_crvStepMargin) continue;
```

Comment:
Is step.start and endTime() wrapped by the microbunchPeriod yet at this point?

Reply:
No, time wrapping happens later after the photons are produced.